src/main/java/org/mozilla/msrp/platform/vertical/video/VideoService.kt

```diff
@@ -20,38 +20,32 @@ class VideoService @Inject constructor(
     private val CACHE_THRESHOLD = 30 * 24 * 60 * 60 * 1000L // a month
 
     fun fromCache(query: String, order: String, maxResult: Long): List<VideoItem> {
-        val cacheKey = query
-
-        val content: String? = videoCacheRepository.get(query)
-        // if not in cache
-        if (content == null) {
-            VerticalMetrics.event(VerticalMetrics.EVENT_CACHE_MISSED, cacheKey)
-            val data = loadData(query, order, maxResult)
-            videoCacheRepository.set(cacheKey, mapper.writeValueAsString(data))
-            return data.videos
-        }
         try {
-            val cache = mapper.readValue(content, VideoServiceResult::class.java)
-            // Use System.currentTimeMillis() means we can only deploy this service in the same timezone.
-            // But it's fine right now.
-            if (System.currentTimeMillis()- cache.ts > CACHE_THRESHOLD) {
-                VerticalMetrics.event(VerticalMetrics.EVENT_CACHE_EXPIRED, cacheKey)
-                val data = loadData(query, order, maxResult)
-                if (data.videos.isNotEmpty()) {
-                    videoCacheRepository.set(cacheKey, mapper.writeValueAsString(data))
-                    return data.videos
+            val content: String? = videoCacheRepository.get(query)
+            if (content != null) {
+                val cache = mapper.readValue(content, VideoServiceResult::class.java)
+                if (cache.videos.isEmpty()) {
+                    VerticalMetrics.event(VerticalMetrics.EVENT_CACHE_MISSED, query)
+                } else if (System.currentTimeMillis() - cache.ts > CACHE_THRESHOLD) {
+                    VerticalMetrics.event(VerticalMetrics.EVENT_CACHE_EXPIRED, query)
+                } else {
+                    return cache.videos
```

Comment:
I'll use the cached content only when
1. there cache video list is not empty
2. the cache is not expired.

other than that, I'll use the new content